rslib/src/search/sqlwriter.rs

```diff
@@ -478,20 +479,34 @@ impl SqlWriter<'_> {
         Ok(())
     }
 
-    fn write_added(&mut self, days: u32) -> Result<()> {
+    fn cutoff_in_secs_from_days(&mut self, days: u32) -> Result<i64> {
         let timing = self.col.timing_today()?;
-        let cutoff = (timing.next_day_at.0 - (86_400 * (days as i64))) * 1_000;
+        Ok(timing.next_day_at.0 - (86_400 * (days as i64)))
+    }
+
+    fn write_added(&mut self, days: u32) -> Result<()> {
+        let cutoff = self.cutoff_in_secs_from_days(days)? * 1_000;
         write!(self.sql, "c.id > {}", cutoff).unwrap();
         Ok(())
     }
 
     fn write_edited(&mut self, days: u32) -> Result<()> {
-        let timing = self.col.timing_today()?;
-        let cutoff = timing.next_day_at.0 - (86_400 * (days as i64));
+        let cutoff = self.cutoff_in_secs_from_days(days)?;
         write!(self.sql, "n.mod > {}", cutoff).unwrap();
         Ok(())
     }
 
+    fn write_introduced(&mut self, days: u32) -> Result<()> {
+        let cutoff = self.cutoff_in_secs_from_days(days)? * 1_000;
+        write!(
+            self.sql,
+            "(select min(id) > {} from revlog where cid = c.id)",
+            cutoff
+        )
+        .unwrap();
```

Comment:
Adding an extra restriction seems to speed things up:

```
sqlite> select count(*) from cards c where (select min(id) > ((strftime('%s','now')-86400*30)*1000)  from revlog where cid = c.id);
count(*) = 94
Run Time: real 0.027 user 0.026584 sys 0.000219
sqlite> select count(*) from cards c where ((select min(id) > ((strftime('%s','now')-86400*30)*1000)  from revlog where cid = c.id) and c.id in (select cid from revlog where id > ((strftime('%s','now')-86400*30)*1000)));
count(*) = 94
Run Time: real 0.005 user 0.003979 sys 0.001189
```

Reply:
That's a tremendous speedup, but I wonder why. 🤔 
Is it that SQL can cache the list of cards in the second statement while it would have to filter the whole revlog table for `cid = c.id` for every row otherwise?

Reply:
Yeah, I believe it means instead of having to do a separate subquery for each card in the collection, it can do one index scan of revlog to fetch candidate card ids, and then do a subquery for only those cards instead.

Thanks as always!